catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -512,7 +512,7 @@ private QueryRequestImpl createQueryRequest(DeleteRequest deleteRequest) {
         QueryImpl queryImpl =
                 new QueryImpl(queryOperations.getFilterWithAdditionalFilters(idFilters),
                         1,  /* start index */
-                        0,  /* page size */
+                        deleteRequest.getAttributeValues().size(),  /* page size */
```

Comment:
❗️ This change may not work in all cases since other code paths may not put the list of metacard IDs in the request.